ql/src/java/org/apache/hadoop/hive/ql/io/orc/OrcSplit.java

```diff
@@ -103,7 +103,7 @@ public OrcSplit(Path path, Object fileId, long offset, long length, String[] hos
     this.isOriginal = isOriginal;
     this.hasBase = hasBase;
     this.rootDir = rootDir;
-    int bucketId = AcidUtils.parseBucketId(path);
+    bucketId = AcidUtils.parseBucketId(path);
```

Comment:
If we only set `bucketId` here then it can be made final but is it safe to do so? I am skeptical cause so far we were setting this variable in the `parse` method. Maybe there is something in the history explaining what it was like that.

Reply:
I didn't find explanation in history why bucket id was set in parse method. Regarding making bucketId final: bucketId is a private class member and cannot be set outside of OrcSplit. Before my change it was set using getPath() method from its parent class. This method uses 'fs' class member from the parent class which is package-private and unaccessible from OrcSplit. So, bucketId is set using path which cannot be modified and it can have only one value. Therefore in my opinion bucketId can be set final and it is safe to do. In other words, if bucketId is set in constructor or in parse method, it can be assigned only same unique value which is based on the path and the path can't be changed at runtime.